ifc_rs/src/lib.rs

```diff
@@ -80,17 +79,25 @@ impl IFC {
     }
 }
 
+
+impl IFCParse for IFC {
+    fn parse<'a>() -> impl parser::IFCParser<'a, Self> {
+        winnow::seq! {
+            IFC {
+                header: Header::parse(),
+                data: DataMap::parse(),
+                footer: Footer::parse(),
+            }
+        }
+    }
+}
+
 impl FromStr for IFC {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self> {
-        let me = seq!(Self {
-            header: Header::parse(),
-            data: DataMap::parse(),
-            footer: Footer::parse(),
-        })
-        .parse(s)
-        .map_err(|err| anyhow!("parsing failed: {err:#?}"))?;
+        let me = s.parse::<IFC>()
```

Comment:
I think this `parse` is resolved to the wrong trait. It's the `core::str::parse` which will internally just call `FromStr` for the turbofished type. This creates an infinite loop and overflows the stack. I see what you wanted to do and I'll fix it myself.

Reply:
Yeah, no worries. I should probably just setup CI for github as well.

Yeah, unfortunately `winnow/debug` is quiet slow. This comes from all the whitespace and comment parsing which is allowed between almost all parsable words. I'd optimize this in the future, but it's not super high on my priority list as of now. 

Some debugging things I tend to do:

- try to keep the individual parsers small and modular and test each one individually before parsing bigger files
- `winnow/debug` can actually be combined with `--release` (only if you didn't already know)